pkg/common/key.go

```diff
@@ -0,0 +1,110 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	guuid "github.com/google/uuid"
+)
+
+const (
+	KEY_DELIMETER = "/"
+	LOCKS         = "_locks"
+	COMMENTS      = "_comments"
+	INTERNAL_DB   = "_"
+)
+
+var prefix string
+
+type Key struct {
+	Prefix    string
+	DBName    string
+	TableName string
+	// the id represents uuid for the rows and id for the comments and locks
+	ID string
```

Comment:
can it UUID ... as it is also the `tablename` and `dbname` don't mean much when you store locks and comments ...

Reply:
`DBName` does mean much for locks and comments. They are per database. 
TableName -? but you can as what are the table in key/value store

Reply:
and you can add then helper functions:

```
func NewKey(prefix, service, dbname, table string, uuid ...string) {
}
```

```
func NewCommentKey(prefix string) {
}
```

```
func NewLockKey(string prefix, service, ...) {
}
````

Reply:
the variable arguments list creates possibilities for an errors, e.g. we will have to check if a function received more than a single var argument, and we will have to return an error, and to check it in all places where we call the functions...
For each type (data, comment and lock) I provide 2 functions with the same internal implementation.

- `func NewCommentKey(commentID string)`
-  `func NewCommentTableKey()`
and btw, if a user calls the first function with commentID == "", it will equal to the second (table key ) function.
Added comments about the functions